### PR TITLE
Tweak: AI - Show upgrade badge on context menu if no subscription [ED-12643]

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

File: modules/ai/assets/js/editor/layout-module.js [hunk 1]
```diff
@@ -58,7 +57,27 @@ export default class Module extends elementorModules.editor.utils.Module {
...
+				<path d="M12 5.25C12.2508 5.25 12.485 5.37533 12.6241 5.58397L16.1703 10.9033L20.5315 7.41435C20.7777
+				7.21743 21.1207 7.19544 21.39 7.35933C21.6592 7.52321 21.7973 7.83798 21.7355 8.14709L19.7355
+				18.1471C19.6654 18.4977 19.3576 18.75 19 18.75H5.00004C4.64253 18.75 4.33472 18.4977 4.26461
+				18.1471L2.2646 8.14709C2.20278 7.83798 2.34084 7.52321 2.61012 7.35933C2.8794 7.19544 3.22241
+				7.21743 3.46856 7.41435L7.82977 10.9033L11.376 5.58397C11.5151 5.37533 11.7493 5.25 12 5.25ZM12
+				7.35208L8.62408 12.416C8.50748 12.5909 8.32282 12.7089 8.1151 12.7411C7.90738 12.7734 7.69566
+				12.717 7.53152 12.5857L4.13926 9.87185L5.61489 17.25H18.3852L19.8608 9.87185L16.4686 12.5857C16.3044
+				12.717 16.0927 12.7734 15.885 12.7411C15.6773 12.7089 15.4926 12.5909 15.376 12.416L12 7.35208Z">
+				</path>
+    		</svg>`;
```
Comment (StyleShit) [inline modules/ai/assets/js/editor/layout-module.js:74]: hmm
is there a way to reuse the SVG from `@elementor/icons`?

File: modules/ai/connect/ai.php [hunk 2]
```diff
@@ -38,6 +38,17 @@ public function get_usage() {
...
 
+	public function get_cached_usage() {
+		$cache_key = 'elementor_ai_usage';
+		$cache_time = 24 * HOUR_IN_SECONDS;
+		$usage = get_site_transient( $cache_key );
+		if ( ! $usage ) {
+			$usage = $this->get_usage();
+			set_site_transient( $cache_key, $usage, $cache_time );
+		}
+		return $usage;
```
Comment (StyleShit) [inline modules/ai/connect/ai.php:49]: Let it breath 😅 

```suggestion
		$usage = get_site_transient( $cache_key );
		
		if ( ! $usage ) {
			$usage = $this->get_usage();
			set_site_transient( $cache_key, $usage, $cache_time );
		}
		
		return $usage;
```

File: modules/ai/module.php [hunk 3]
```diff
@@ -148,6 +148,8 @@ private function enqueue_main_script() {
 			[
 				'is_get_started' => User::get_introduction_meta( 'ai_get_started' ),
 				'connect_url' => $this->get_ai_connect_url(),
+				// Use a cached version, don't call the API on every editor load.
+				'usage' => $this->get_ai_app()->get_cached_usage(),
```
Comment (StyleShit) [inline modules/ai/module.php:152]: 🔥

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: modules/ai/assets/js/editor/layout-module.js (hunk 1 above)
Comment (matipojo) [reply modules/ai/assets/js/editor/layout-module.js:74]: It's outside of React.

**Review by StyleShit — DISMISSED** [review]

**Review by StyleShit — APPROVED** [review]

**Pull request merged**